pkg/platform/kube/platform.go

```diff
@@ -1263,6 +1313,22 @@ func (p *Platform) validateAPIGatewayFunctionsHaveNoIngresses(apiGatewayConfig *
 	return errGroup.Wait()
 }
 
+func (p *Platform) validateFunctionIngresses(functionConfig *functionconfig.Config) error {
+	if err := p.validateFunctionNoIngressAndAPIGateway(functionConfig); err != nil {
+		return errors.Wrap(err, "Failed while validating function is not exposed both by internal ingresses and api gateway")
+	}
+
+	for _, ingressInstance := range functionconfig.GetIngressesFromTriggers(functionConfig.Spec.Triggers) {
+		for _, path := range ingressInstance.Paths {
+			if err := p.validateIngressHostAndPathAvailability(functionConfig.Meta.Namespace, ingressInstance.Host, path); err != nil {
```

Comment:
🔔 🔔
Each one of this call, has `List` operation on *all* namespace ingresses.

1. use errGroup where possible
2. aggregate all `ingressInstances` and Paths into a single request
3. List ingresses once

Reply:
done